command/operator_diagnose.go

```diff
@@ -284,7 +282,238 @@ func (c *OperatorDiagnoseCommand) offlineDiagnostics(ctx context.Context) error
 			if err != nil {
 				return err
 			}
+
+			// Initialize the Service Discovery, if there is one
+			configSR, err = beginServiceRegistration(server, config)
+			if err != nil {
+				return err
+			}
 		}
 		return nil
-	})
+	}); err != nil {
+		diagnose.Error(ctx, err)
+	}
+
+	var barrierWrapper *wrapping.Wrapper
+	var barrierSeal *vault.Seal
+	var unwrapSeal *vault.Seal
+	if err := diagnose.Test(ctx, "create-seal", func(context.Context) error {
+		var seals []vault.Seal
+		var sealConfigError error
+		bS, bW, uS, seals, sealConfigError, err := setSeal(server, config, make([]string, 0), make(map[string]string))
+		// Check error here
+		if err != nil {
+			return err
+		}
+		if sealConfigError != nil {
+			return fmt.Errorf("seal could not be configured: seals may already be initialized")
+		}
+
+		barrierSeal = &bS
+		barrierWrapper = &bW
+		unwrapSeal = &uS
+		if seals != nil {
+			for _, seal := range seals {
+				// Ensure that the seal finalizer is called, even if using verify-only
```

Comment:
This is maybe a good place to add another span with the seal type.

Reply:
Nah, sorry, not for the finalize part.  In the next effort where we test each seal, we might want that.